### PR TITLE
PyFloat_AS_DOUBLE()

### PR DESCRIPTION
**Pull request by ijl**
This is a reimplementation of https://github.com/python/cpython/blob/e42b705188271da108de42b55d9344642170aa2b/Include/floatobject.h#L50-L52.

### CODE REVIEW

File: src/ffi3/floatobject.rs [hunk 1]
```diff
@@ -1,6 +1,12 @@
 use crate::ffi3::object::*;
 use std::os::raw::{c_double, c_int};
 
+#[repr(C)]
```
Comment (kngwyu) [inline src/ffi3/floatobject.rs]: We also need `#[cfg(not(Py_LIMITED_API))]` here

File: src/types/floatob.rs [hunk 2]
```diff
@@ -103,4 +103,14 @@ mod test {
...
+
+    #[test]
+    fn test_as_double_macro() {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let v = 1.23f64;
+        let obj = v.to_object(py);
+        assert_eq!(v, unsafe { PyFloat_AS_DOUBLE(obj.as_ptr()) });
+    }
+
```
Comment (kngwyu) [inline src/types/floatob.rs]: Just a nit but please remove this empty line.

**Comment by ijl** [comment]
@kngwyu ok, resolved.

**Comment by ijl** [comment]
The last two pushes have had failures on "Collecting coverage data". master is affected in the same way.

**Comment by althonos** [comment]
@ijl : yes, my PR was successful but the same commit when merged in master was not because of this. I tried deleting the cache but it didn't change anything. Will try to debug that.

**Comment by ijl** [comment]
Ok, well can this merge to master?

**Comment by kngwyu** [comment]
Thanks.
I think we should fix the coverage problem, but I don't think we should wait for the fix, since this PR is not related to it.

**Pull request merged**